### PR TITLE
Create new `cli` library with a simplified interface

### LINKED ISSUE #552: Tracking Issue: `sov-cli` improvements
## Background
`sov-cli` grew organically from a simple transaction generator for the bank module into a complex command line tool. Unfortunately, this organic growth has made it more and more unwieldy over time. At Modular summit, one of the biggest pieces of feedback we received was that the cli was unintuitive and hard to use. This is completely fair. Now that #423, #525, and #446  have landed, we have the scaffolding to re-architect the CLI into a standalone crate and provide a much better UX. 

## Tasks
- [x] Move sov-cli into its own crate
- [ ] Refactor `sov-cli` interface to make it easier to use
- [x] Add persistance (i.e. for private keys) to `sov-cli`
- [x] Add RPC capabilities to `sov-cli`

### PR DESCRIPTION
**Pull request by preston-evans98**
# Description
This PR creates a new `sov-cli` library and a simple binary which combines the library and the `demo-stf` to create a simple CLI. 

The new CLI allows you to generate and persist accounts, and switch between active accounts using a simple interface:

```
Usage: main keys <COMMAND>

Commands:
  generate  Generate a new key pair
  import    Import an existing key pair
  list      List the keys in this wallet
  activate  Set the active key
  help      Print this message or the help of the given subcommand(s)
```
It also provides an API for creating batches of transactions either by providing args on the command line (using `clap`) or by providing pre-generated JSON. 
```
$ cargo run transactions
Commands:
  generate  Parse a transaction from command-line arguments
  import    Import a pre-formatted transaction from a JSON file or as a JSON string
  list      List the current batch of transactions
  help      Print this message or the help of the given subcommand(s)

Options:
  -h, --help  Print help
```

To keep this PR small, several changes are left as TODOs:
- Add the ability to send transactions to a full node
- Add the ability to drop/update the current batch of transactions
- Add a command to generate json schemas
- Add the ability to auto-fill nonces
- Replace the old `sov-cli` binary in demo-stf with a new one based on this library.
  
## Linked Issues
- Related to #552

### CODE REVIEW

File: module-system/sov-cli/src/bin/main.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
+use demo_stf::runtime::{Runtime, RuntimeCall};
```
Comment (bkolad) [inline module-system/sov-cli/src/bin/main.rs:1]: let's add
```Rust
#![deny(missing_docs)]
#![doc = include_str!("../README.md")]

```

File: module-system/sov-cli/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+use std::env;
```
Comment (bkolad) [inline module-system/sov-cli/src/lib.rs:3]: let's add:
```Rust
#![deny(missing_docs)]
#![doc = include_str!("../README.md")]

File: module-system/sov-cli/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+pub mod wallet_state;
+pub mod workflows;
+
+const SOV_WALLET_DIR_ENV_VAR: &str = "SOV_WALLET_DIR";
+
+/// The directory where the wallet is stored.
+pub fn wallet_dir() -> Result<impl AsRef<Path>, anyhow::Error> {
+    // First try to parse from the env variable
+    if let Ok(val) = env::var(SOV_WALLET_DIR_ENV_VAR) {
+        return PathBuf::try_from(val).map_err(|e: std::convert::Infallible| {
```
Comment (bkolad) [inline module-system/sov-cli/src/lib.rs:16]: Why do we `map_err` here if it can never happen: `std::convert::Infallible`?

Comment (bkolad) [inline module-system/sov-cli/src/lib.rs]: Why do we `map_err` on `std::convert::Infallible` since it can never happen?

**Comment by bkolad** [comment]
Question: Does the doc in `https://github.com/Sovereign-Labs/sovereign-sdk/tree/nightly/examples/demo-rollup` also require an update?

**Comment by preston-evans98** [comment]
> Question: Does the doc in `https://github.com/Sovereign-Labs/sovereign-sdk/tree/nightly/examples/demo-rollup` also require an update?

Good question! It will, but I'll make that update in the change where I remove the old cli.

File: module-system/sov-cli/src/lib.rs (hunk 3 above)
Comment (preston-evans98) [reply module-system/sov-cli/src/lib.rs]: Oops! Haha I let my IDE fill in that type annotation. Switching to `val.into()`.

File: module-system/sov-cli/src/lib.rs (hunk 2 above)
Comment (preston-evans98) [reply module-system/sov-cli/src/lib.rs:3]: Done!

**Review by bkolad — APPROVED** [review]

**Pull request merged**